src/game/objects/har.h

```diff
@@ -153,11 +153,15 @@ typedef struct har_t {
     uint8_t delay; // used for 'stretching' frames in netplay
 
     // ptr, pe, etc. stuff
-    uint8_t p_pal_ref;
-    uint8_t p_har_switch;
-    uint8_t p_color_ref;
-    int p_ticks_left;
-    unsigned int p_ticks_length;
+    uint8_t p_pal_ref;    // "pd"
+    uint8_t p_har_switch; // "pe"
+    int16_t p_fade_out_ticks;
+    int16_t p_fade_out_ticks_left;
+    int16_t p_fade_in_ticks;
+    int16_t p_fade_in_ticks_left;
+    int16_t p_sustain_ticks_left;
```

Comment:
Why aren't these in object/player.c?

Reply:
I'm not sure why they were put here in 2014, but I left them here now because they are not bound to the current state of the animation string -- instead, they should work even if the string gets changed. I'm not entirely sure what the best place would be.

I think the whole player model thing is wrong, but not sure what the correct way would be.